bson/bson_test.go

```diff
@@ -111,3 +111,17 @@ func TestD(t *testing.T) {
 		}
 	})
 }
+
+func TestExtJSONEscapeKey(t *testing.T) {
+	doc := D{{Key: "\"", Value: 1}}
+	b, err := MarshalExtJSON(&doc, false, false)
+	noerr(t, err)
+
+	want := "{\"\\\"\":1}"
+	if diff := cmp.Diff(want, string(b)); diff != "" {
+		t.Errorf("Marshaled documents do not match. got %v, want %v", string(b), want)
+	}
+
+	err = UnmarshalExtJSON(b, false, &doc)
```

Comment:
To also ensure that the driver is unescaping the keys correctly, unmarshal into a different document and assert that it's equals to `doc`.

Reply:
done